presto-main/src/main/java/com/facebook/presto/sql/planner/TranslationMap.java

```diff
@@ -258,12 +258,14 @@ private Expression rewriteExpressionWithResolvedName(Expression node)
             @Override
             public Expression rewriteDereferenceExpression(DereferenceExpression node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
             {
-                Optional<ResolvedField> resolvedField = rewriteBase.getScope().tryResolveField(node);
-                if (resolvedField.isPresent()) {
-                    if (resolvedField.get().isLocal()) {
-                        return getSymbol(rewriteBase, node)
-                                .map(symbol -> coerceIfNecessary(node, symbol.toSymbolReference()))
-                                .orElseThrow(() -> new IllegalStateException("No symbol mapping for node " + node));
+                if (analysis.isColumnReference(node)) {
+                    Optional<ResolvedField> resolvedField = rewriteBase.getScope().tryResolveField(node);
+                    if (resolvedField.isPresent()) {
```

Comment:
I think this should be check state: `checkState(resolvedField.isPresent()`) as column references should be resolvable by corresponding scope

Reply:
I think they don't need to be. The check was here before and i think i tried to remove it..

Reply:
It would be strange if the `if (resolvedField.isPresent()) {` is needed for tests to pass. If the expression is column reference then the scope here should be the one that resolved it during analysis. Subqueries should be translated already and lambda scope doesn't resolve column references at all.

If the scope here is different than the one that resolved column reference during analysis then it's potentially semantically incorrect. Such different scope might not resolve field, but it might also (incorrectly) resolve field to a wrong field index.

In any case, we need to investigate if this issue is introduced by this PR or was already in the code. It could be that the `if (resolvedField.isPresent()) {` is required and correct because it's just some edge case (e.g: part of ORDER BY support or some subquery is not supported and wasn't rewritten).

Reply:
There is an edge-case related to order by planning in `QueryPlanner`, needs to stay